packages/client/test/l2_e2e.spec.ts

```diff
@@ -0,0 +1,159 @@
+  ).deploy(gatewayUrl, signers)
+
+  offchainResolver = await eth.getContractAt(
+    abiOffchainResolver,
+    await ResolverContract.getAddress(),
+  )
+  console.log('Offchain resolver: ', await ResolverContract.getAddress())
+}
+
+// Function to deploy registry contract
+async function deployRegistry(): Promise<void> {
+  const l1Signer = await l1Provider.getSigner(devAccountPublicKey)
+  const RegistryContract = await new eth.ContractFactory(
+    abiRegistry,
+    bytecodeRegistry,
+    l1Signer,
+  ).deploy()
+
+  registry = await eth.getContractAt(
+    abiRegistry,
+    await RegistryContract.getAddress(),
+  )
+  console.log('Registry: ', await RegistryContract.getAddress())
+
+  await (registry.connect(l1Signer) as Contract).setSubnodeRecord(
+    root,
+    labelhash('eth'),
+    l1Signer,
+    await offchainResolver.getAddress(),
+    10000000,
+  )
+  await (registry.connect(l1Signer) as Contract).setSubnodeRecord(
+    namehash('eth'),
+    labelhash('offchain'),
+    l1Signer,
+    await offchainResolver.getAddress(),
+    10000000,
+  )
+}
+
+// Function to deploy universal resolver contract
+async function deployUniversalResolver(): Promise<void> {
+  const l1Signer = await l1Provider.getSigner(devAccountPublicKey)
+
+  UniversalResolverContract = await new eth.ContractFactory(
+    abiUniversalResolver,
+    bytecodeUniversalResolver,
+    l1Signer,
+  ).deploy(await registry.getAddress(), [gatewayUrl])
+
+  universalResolver = await eth.getContractAt(
+    abiUniversalResolver,
+    await UniversalResolverContract.getAddress(),
+  )
+  console.log('universal resolver: ', await universalResolver.getAddress())
+}
+
+describe('SeuContrato', () => {
+  before(async () => {
+    const offSigs = await l2Provider.listAccounts()
+    console.log(offSigs)
+    const s = await l1Provider._detectNetwork()
+    console.log(s)
+
+    console.log('Inside before.')
+    signers = await eth.getSigners()
+    // Deploying the contracts
+    await deployOffchainResolver()
+    await deployRegistry()
+    await deployUniversalResolver()
+  })
+
+  /*
+   This test will fail when it reach the gateway
+   because the offchain part is not implemented yet.
+  */
+  it('Call ENS flow with viem.', async () => {
+    // ENS address
+    const ensAddress = normalize('public.eth')
+    // Getting Avatar from the ens address
+    const client = createTestClient({
+      chain: localhost,
+      mode: 'hardhat',
+      transport: http('http://127.0.0.1:8545'),
+    }).extend(publicActions)
+
+    //   console.log('ok!')
+    //   try {
+    //     await client.getEnsAvatar({
+    //       name: ensAddress,
+    //       universalResolverAddress:
+    //         (await UniversalResolverContract.getAddress()) as `0x${string}`,
+    //     })
+    //   } catch (error) {
+    //     const customError = error as BaseError
+    //     console.log(customError.message)
+    //     // expect(customError.message).contain('HTTP request failed')
+    //   }
```

Comment:
is this commented out on purpose?

Reply:
It will be used after the gatway_l2 implementation. That is why it was not removed.